ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/QueryCapacityUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.ksql.KsqlEngine;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+
+public final class QueryCapacityUtil {
+  private QueryCapacityUtil() {
+  }
+
+  public static boolean exceedsPersistentQueryCapacity(
+      final KsqlEngine ksqlEngine, final KsqlConfig ksqlConfig, final long additionalQueries) {
+    return ksqlEngine.numberOfPersistentQueries() + additionalQueries > getQueryLimit(ksqlConfig);
```

Comment:
nit: consider adding parenthesis so its unambiguous when operators take precedence. 
  
```suggestion
    return (ksqlEngine.numberOfPersistentQueries() + additionalQueries) > getQueryLimit(ksqlConfig);
```